lib/tenderjit/iseq_compiler.rb

```diff
@@ -1786,6 +1786,19 @@ def handle_adjuststack n
       n.times { @temp_stack.pop }
     end
 
+    def handle_tostring
+      address = Fiddle::Handle::DEFAULT["rb_obj_as_string_result"]
+      val = @temp_stack.pop
+      str = @temp_stack.pop
+
+      with_runtime do |rt|
+        rt.push_reg REG_BP
+        new_string_address = rt.call_cfunc_without_alignment(address, [val, str])
+        rt.push new_string_address, name: "string"
```

Comment:
`rt.push` is pushing on to the temporary stack.  You should be able to call the cfunc _with_ alignment and then safely push on to the compile-time temp stack